src/api/v2/types.rs

```diff
@@ -335,3 +356,36 @@ pub fn handle_result(result: Result<impl Reply, impl Reply>) -> impl Reply {
 		Err(err) => err.into_response(),
 	}
 }
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "topic", rename_all = "kebab-case")]
+pub enum WsResponse {
+	Error(Error),
+	Version(Response<Version>),
+	Status(Response<Status>),
+	DataTransactionSubmitted(Response<SubmitResponse>),
+}
+
+impl From<Response<Version>> for WsResponse {
+	fn from(value: Response<Version>) -> Self {
+		WsResponse::Version(value)
+	}
+}
+
+impl From<Response<Status>> for WsResponse {
+	fn from(value: Response<Status>) -> Self {
+		WsResponse::Status(value)
+	}
+}
+
+impl From<Response<SubmitResponse>> for WsResponse {
+	fn from(value: Response<SubmitResponse>) -> Self {
+		WsResponse::DataTransactionSubmitted(value)
+	}
+}
+
+impl From<Error> for WsResponse {
+	fn from(value: Error) -> Self {
+		WsResponse::Error(value)
+	}
+}
```

Comment:
I think you can introduce `derive_more` crate and derive `From` and also `TryFrom` for each enum variant.